src/mgmt/cli_mgmt.c

```diff
@@ -536,13 +537,35 @@ dc_mgmt_put_sys_info(struct daos_sys_info *info)
 
 #define SYS_INFO_BUF_SIZE 16
 
-static int g_num_serv_ranks = 1;
+static int g_num_serv_ranks = -1;
+static d_rank_t *g_serv_ranks;
 
-int dc_mgmt_net_get_num_srv_ranks(void)
+/* Return the number of attached ranks.  */
+int
+dc_mgmt_net_get_num_srv_ranks(void)
 {
+	D_ASSERT(g_num_serv_ranks >= 0);
+
 	return g_num_serv_ranks;
 }
 
+/* Return the rank id of an attached rank.  */
+int
+dc_mgmt_net_get_srv_rank(int idx, d_rank_t *rank)
+{
+	D_ASSERT(rank != NULL);
+	D_ASSERT(g_num_serv_ranks >= 0);
+
+	if (idx >= g_num_serv_ranks) {
+		D_ERROR("Invalid rank index: index=%d, ranks_num=%d\n", idx, g_num_serv_ranks);
+		return -DER_NONEXIST;
+	}
+
+	*rank =  g_serv_ranks[idx];
+
+	return -DER_SUCCESS;
+}
```

Comment:
We have a special CRT_NO_RANK rank value to indicate errors. With that you can eliminate need for separate rc with something like:
d_rank_t dc_mgmt_net_get_srv_rank(int idx) {
    return (idx < g_num_serv_ranks) ? g_serv_ranks[idx] : CRT_NO_RANK;
}

Reply:
> Did not know that special value. I will change it according to your comments.
> 
> * [x]  Use of `CRT_NO_RANK` for error management

Fixed with commit c8b0636